Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyDraftOrdersAPI.Codeunit.al

```diff
@@ -0,0 +1,385 @@
+
+                GraphQuery.Append(', originalUnitPrice :');
+                GraphQuery.Append(Format(TempShpfyOrderLine."Unit Price", 0, 9));
+
+                GraphQuery.Append('},');
+            until TempShpfyOrderLine.Next() = 0;
+
+            foreach TaxTitle in ShpfyOrderTaxLines.Keys() do begin
+                GraphQuery.Append('{');
+                GraphQuery.Append('title: \"');
+                GraphQuery.Append(ShpfyCommunicationMgt.EscapeGrapQLData(TaxTitle));
+                GraphQuery.Append('\"');
+
+                GraphQuery.Append(', quantity: ');
+                GraphQuery.Append(Format(1, 0, 9));
+
+                GraphQuery.Append(', originalUnitPrice: ');
+                GraphQuery.Append(Format(ShpfyOrderTaxLines.Get(TaxTitle), 0, 9));
+
+                GraphQuery.Append('},');
+            end;
+            GraphQuery.Remove(GraphQuery.Length(), 1);
+        end;
+        GraphQuery.Append(']');
+    end;
+
+    local procedure AddDiscountAmountToGraphQuery(var GraphQuery: TextBuilder; DiscountAmount: Decimal; DiscountTitle: Text)
+    begin
+        GraphQuery.Append(', appliedDiscount: {');
+        GraphQuery.Append('description: \"');
+        GraphQuery.Append(ShpfyCommunicationMgt.EscapeGrapQLData(DiscountTitle));
+        GraphQuery.Append('\"');
+
+        GraphQuery.Append(', value: ');
+        GraphQuery.Append(Format(DiscountAmount, 0, 9));
+
+        GraphQuery.Append(', valueType: ');
+        GraphQuery.Append('FIXED_AMOUNT');
+
+        GraphQuery.Append(', title: \"');
+        GraphQuery.Append(ShpfyCommunicationMgt.EscapeGrapQLData(DiscountTitle));
+        GraphQuery.Append('\"');
+
+        GraphQuery.Append('}');
+    end;
+
+    local procedure AddNote(var GraphQuery: TextBuilder; var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary)
+    var
+        SalesCommentLine: Record "Sales Comment Line";
+        NotesTextBuilder: TextBuilder;
+    begin
+        SalesCommentLine.SetRange("Document Type", SalesCommentLine."Document Type"::"Posted Invoice");
+        SalesCommentLine.SetRange("No.", TempShpfyOrderHeader."Sales Invoice No.");
+
+        if SalesCommentLine.FindSet() then begin
+            GraphQuery.Append(', note: \"');
+            repeat
+                NotesTextBuilder.Append(SalesCommentLine.Comment + '\n');
+            until SalesCommentLine.Next() = 0;
+            GraphQuery.Append(NotesTextBuilder.ToText());
+            GraphQuery.Append('\"');
+        end;
+    end;
+
+    local procedure AddPaymentTerms(var GraphQuery: TextBuilder; var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary)
+    var
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        ShpfyPaymentTerm: Record "Shpfy Payment Terms";
+        DueAtDateTime: DateTime;
+        IssuedAtDateTime: DateTime;
+    begin
+        if not ShopifyPaymentTermsExists(ShpfyPaymentTerm, TempShpfyOrderHeader, SalesInvoiceHeader) then
```

Comment:
I would verify this with Andrei. I just sent a couple of invoices and all went through, but because I didn't have payment terms mapped, they went through successfully, but they were marked as "Paid" even though they shouldn't. To me the sending should fail, or revert to some default payment terms that are always mapped, but check with Andrei